timechop/timechop.py

```diff
@@ -25,21 +26,29 @@ def chop_time(self):
                 )
         return(matrix_set_definitions)
 
-    def calculate_matrix_end_times(self):
+    def calculate_matrix_end_times(self, look_back_duration):
         matrix_end_times = []
         update_delta = utils.convert_str_to_relativedelta(self.update_window)
         matrix_end_time = self.modeling_end_time - update_delta
-        while matrix_end_time > self.modeling_start_time:
+        look_back_delta = utils.convert_str_to_relativedelta(look_back_duration)
+        matrix_start_time = matrix_end_time - look_back_delta
+        if matrix_start_time < self.modeling_start_time:
+            raise ValueError('''
+                Start of first matrix, {}, is before modeling start time, {}.
+            '''.format(matrix_start_time, self.modeling_start_time))
+
+        while matrix_start_time >= self.modeling_start_time:
             matrix_end_times.insert(0, matrix_end_time)
             matrix_end_time -= update_delta
+            matrix_start_time = matrix_end_time - look_back_delta
 
-        if matrix_end_time == self.modeling_start_time:
-            return(matrix_end_times)
-        else:
-            raise ValueError('''Modeling period not evenly divisbile by update
+        if matrix_end_time != self.modeling_start_time:
+            warnings.warn('''Modeling period not evenly divisbile by update
```

Comment:
I've only recently started using the warnings model myself (a few days ago because I specifically wanted to raise a DeprecationWarning) and am trying to figure out when it's suitable to use it versus a simple logging.warning.

From [this page](http://stackoverflow.com/questions/9595009/python-warnings-warn-vs-logging-warning), 
```

    warnings.warn() in library code if the issue is avoidable and the client application should be modified to eliminate the warning

    logging.warning() if there is nothing the client application can do about the situation, but the event should still be noted
```
I started out thinking that logging.warning was actually more suitable here, but after thinking about this a bit I like warnings.warn here. We are telling the client application (i.e., a particular project) that their temporal configuration may have some problems and can be improved.